applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -318,6 +318,55 @@ void NavierStokesWallCondition<3,3>::GetDofList(DofsVectorType& rElementalDofLis
     }
 }
 
+template<unsigned int TDim, unsigned int TNumNodes>
+void NavierStokesWallCondition<TDim, TNumNodes>::Calculate(
+    const Variable< array_1d<double,3> >& rVariable,
+    array_1d<double,3>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    rOutput = ZeroVector(3);
+
+    if (rVariable == DRAG_FORCE) {
+        const auto& r_geom = GetGeometry();
+        const auto& r_integration_points = r_geom.IntegrationPoints(GeometryData::GI_GAUSS_2);
+        unsigned int n_gauss = r_integration_points.size();
+        Vector det_J_vect = ZeroVector(n_gauss);
+        r_geom.DeterminantOfJacobian(det_J_vect, GeometryData::GI_GAUSS_2);
+        const auto N_container = r_geom.ShapeFunctionsValues(GeometryData::GI_GAUSS_2);
+
+        // Calculate normal
+        array_1d<double,3> normal;
+        CalculateNormal(normal);
+        normal /= norm_2(normal);
+
+        // Finding parent element to retrieve viscous stresses
+        // Note that we assume in here that the shear stress is constant inside the element
+        auto& r_neighbours = this->GetValue(NEIGHBOUR_ELEMENTS);
+        KRATOS_ERROR_IF(r_neighbours.size() > 1) << "A condition was assigned more than one parent element." << std::endl;
```

Comment:
Just for understanding: 
Why are the neighbor elements the parent element?

Reply:
This is what we decided long time ago. The `NEIGBOUR_ELEMENTS` of a condition contains its parent.

Reply:
Cannot we use a different variable for this?
What if a condition needs its neighbors?

Reply:
> Cannot we use a different variable for this?
> What if a condition needs its neighbors?

For that we have the `NEIGHBOUR_CONDITIONS` variable.

Reply:
Thanks!

Reply:
hm ok but what was the reason not to use e.g. `PARENT_ELEMENTS`? Just curious

Reply:
Honestly, I can't remember...